warehouse/query-core/src/main/java/datawave/query/attributes/Attribute.java

```diff
@@ -41,6 +43,9 @@ public abstract class Attribute<T extends Comparable<T>> implements WritableComp
     protected int hashcode = Integer.MIN_VALUE;
     protected long sizeInBytes = Long.MIN_VALUE;
 
+    // used by Document, Attributes and TypeAttribute
+    protected static final ThreadLocal<ClassCache> classCache = ThreadLocal.withInitial(CaffeineClassCache::new);
```

Comment:
Just curious why its thread local. Off the top of my head it sounds like it'd be better to share across all threads since we're still gonna end up doing reflection on each thread. Is it to remove contention?